src/core/qgsmaplayer.cpp

```diff
@@ -2189,6 +2191,75 @@ void QgsMapLayer::invalidateWgs84Extent()
   mWgs84Extent = QgsRectangle();
 }
 
+QString QgsMapLayer::generalHtmlMetadata() const
+{
+  QString metadata = QStringLiteral( "<h1>" ) + tr( "General" ) + QStringLiteral( "</h1>\n<hr>\n" ) + QStringLiteral( "<table class=\"list-view\">\n" );
+
+  // name
+  metadata += QStringLiteral( "<tr><td class=\"highlight\">" ) + tr( "Name" ) + QStringLiteral( "</td><td>" ) + name() + QStringLiteral( "</td></tr>\n" );
+
+  QString path;
+  bool isLocalPath = false;
+  if ( dataProvider() )
+  {
+    // local path
+    QVariantMap uriComponents = QgsProviderRegistry::instance()->decodeUri( dataProvider()->name(), publicSource() );
+    if ( uriComponents.contains( QStringLiteral( "path" ) ) )
+    {
+      path = uriComponents[QStringLiteral( "path" )].toString();
+      QFileInfo fi( path );
+      if ( fi.exists() )
+      {
+        isLocalPath = true;
+        metadata += QStringLiteral( "<tr><td class=\"highlight\">" ) + tr( "Path" ) + QStringLiteral( "</td><td>%1" ).arg( QStringLiteral( "<a href=\"%1\">%2</a>" ).arg( QUrl::fromLocalFile( path ).toString(), QDir::toNativeSeparators( path ) ) ) + QStringLiteral( "</td></tr>\n" );
+
+        qint64 fileSize = fi.size();
```

Comment:
what happens here for non-file sources? (e.g. a .gdb folder). Should the `fi.isFile()` test be moved up before this?

Reply:
The size returns 0. I guess we should avoid printing size for folder source. One could calculate size of files within folder but let's say for some reason people drag n dropped non-gdb files into their gdb folder, we'd end up with a bad total size.